src/main/java/com/wultra/app/enrollmentserver/errorhandling/DefaultExceptionHandler.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * PowerAuth Enrollment Server
+ * Copyright (C) 2019 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.wultra.app.enrollmentserver.errorhandling;
+
+import io.getlime.core.rest.model.base.response.ErrorResponse;
+import io.getlime.security.powerauth.rest.api.base.exception.PowerAuthAuthenticationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Exception handler for RESTful API issues.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@ControllerAdvice
+public class DefaultExceptionHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(DefaultExceptionHandler.class);
+
+    /**
+     * Default exception handler, for unexpected errors.
+     * @param t Throwable.
+     * @return Response with error details.
+     */
+    @ExceptionHandler(Throwable.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public @ResponseBody ErrorResponse handleDefaultException(Throwable t) {
+        logger.error("Error occurred when processing the request.", t);
+        return new ErrorResponse("ERROR_GENERIC", "Unknown error occurred while processing request.");
+    }
+
+    /**
+     * Exception handler for invalid request exception.
+     * @param ex Exception.
+     * @return Response with error details.
+     */
+    @ExceptionHandler(InvalidRequestObjectException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public @ResponseBody ErrorResponse handleAuthStepException(InvalidRequestObjectException ex) {
```

Comment:
Method name `handleAuthStepException` does not correspond to `InvalidRequestObjectException`.